controllers/profiles/status_enricher_dev.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package profiles
+
+import (
+	"context"
+	"fmt"
+
+	openshiftv1 "github.com/openshift/api/route/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	klabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"knative.dev/pkg/apis"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorapi "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+)
+
+func defaultDevStatusEnricher(ctx context.Context, client client.Client, config *rest.Config, workflow *operatorapi.KogitoServerlessWorkflow) (client.Object, error) {
+	//If the workflow Status hasn't got a NodePort Endpoint, we are ensuring it will be set
+	// If we aren't on OpenShift we will enrich the status with 2 info:
+	// - Address the service can be reached
+	// - Node port used
+	service := &v1.Service{}
+	err := client.Get(ctx, types.NamespacedName{Namespace: workflow.Namespace, Name: workflow.Name}, service)
+	if err != nil {
+		return nil, err
+	}
+	//If the service has got a Port that is a nodePort we have to use it to create the workflow's NodePort Endpoint
+	if service.Spec.Ports != nil && len(service.Spec.Ports) > 0 {
+		if port := findNodePortFromPorts(service.Spec.Ports); port > 0 {
+			clientset, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				return nil, err
+			}
+
+			labelSelector := metav1.LabelSelector{MatchLabels: labels(workflow)}
+			listOptions := metav1.ListOptions{
+				LabelSelector: klabels.Set(labelSelector.MatchLabels).String(),
+			}
+			pods, _ := clientset.CoreV1().Pods(workflow.Namespace).List(ctx, listOptions)
+			var ipaddr string
+			for _, p := range pods.Items {
```

Comment:
I am not sure if we really need it, what is the purpose of getting the Pod's internal IP address?
NodePort uses the node's exposed IP, no?

I mean, from outside the cluster we can't access the pod ip directly.